### PR TITLE
ci(test): store junit results for e2e tests

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Junit summary can be used to give a clean output in CircleCI of which tests have failed

**Result**

- adds @wdio/junit-reporter
- sets up junit report on CircleCI

example for the report:

- passing: https://app.circleci.com/pipelines/github/algolia/instantsearch.js/6322/workflows/0e52b9b3-35be-4f17-ab65-de7d08d5fc78/jobs/30033/tests
- failing: https://app.circleci.com/pipelines/github/algolia/instantsearch.js/6323/workflows/e96ac99e-beba-449d-82a4-d0e250257486/jobs/30039/tests

### CODE REVIEW

File: scripts/website/build-examples.sh [hunk 1]
```diff
@@ -7,11 +7,21 @@ set -e # exit when error
...
-    name=$(basename "$example")
-    echo "Building $name example..."
+build_example() {
+  name=$1
+  echo "Building $name example..."
 
-    (cd "$example" && yarn && yarn build && cp -r "dist/" "../../website/examples/$name/")
-  fi
-done
+  (cd "$example" && yarn && yarn build && cp -r "dist/" "../../website/examples/$name/")
```
Comment (tkrugg) [inline scripts/website/build-examples.sh:14]: does this work? `$example` is not defined here I think.

Comment (Haroenv) [reply scripts/website/build-examples.sh]: you're right, I forgot to copy the right version: 24a4f68e

File: jest.config.js [hunk 2]
```diff
@@ -25,4 +25,18 @@ module.exports = {
...
     __DEV__: true,
   },
+
+  // reporter for circleci
+  reporters: [
+    'default',
+    [
+      'jest-junit',
+      {
+        outputDirectory: 'junit/jest',
```
Comment (tkrugg) [inline jest.config.js]: I find it strange that the e2e equivalent of this file `wdio.saucelabs.conf.js` is adds `__dirname` to `outputDirectory`. I'm not familiar enough to know if it's a typo or done on purpose.

Comment (Haroenv) [reply jest.config.js]: they just use different configuration formats, jest assumes surname, while wdio wants an absolute path

File: scripts/website/build-examples.sh [hunk 3]
```diff
@@ -7,11 +7,26 @@ set -e # exit when error
 rm -rf website/examples
 mkdir website/examples
 
-for example in examples/*; do
-  if [ -d "$example" ]; then
-    name=$(basename "$example")
-    echo "Building $name example..."
+build_example() {
+  example=$1
+  name=$2
```
Comment (tkrugg) [inline scripts/website/build-examples.sh:12]: I would have just moved `name=$(basename "$example")` in here in that case.
otherwise looks good 👍

**Review by tkrugg — APPROVED** [review]

**Pull request merged**